arch/riscv/core/smp.c

```diff
@@ -46,7 +46,7 @@ void z_riscv_secondary_cpu_init(int hartid)
 #ifdef CONFIG_SMP
 	_kernel.cpus[cpu_num].arch.online = true;
 #endif
-#ifdef CONFIG_THREAD_LOCAL_STORAGE
+#if defined(CONFIG_MULTITHREADING) && defined(CONFIG_THREAD_LOCAL_STORAGE)
```

Comment:
how the TLS is related to multithreading?

Reply:
z_idle_threads defined at here.
It is only available in case of MULTITHREADING enabled.

https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/init.c#L51-L53

Since, We need to remove dependency to it if disabled MULTITHREADING.

Reply:
odd. But nothing to do with this PR.